tests/flows/basic_send_receive.go

```diff
@@ -94,7 +95,17 @@ func BasicSendReceive(network interfaces.Network) {
 	//
 	// Relay the message to the destination
 	//
-	network.RelayMessage(ctx, receipt, subnetBInfo, subnetAInfo, true)
+	deliveryReceipt = network.RelayMessage(ctx, receipt, subnetBInfo, subnetAInfo, true)
+
+	// Check that the receipt was received for expected Teleporter message ID
+	// This check is not performed for external networks because unrelated messages may have already changed
+	// the state of the receipt queues.
+	if !network.IsExternalNetwork() {
+		Expect(utils.CheckReceiptReceived(
```

Comment:
In order to not depend on the order that tests run, I think we should clear the receipt queue at the start of this flow. Otherwise, other tests may send messages that cause the receipt queue length to be >5 at the start of this flow, and the first message sent back may not have this specific receipt.

Reply:
moved the `ClearReceiptQueue` into a utils function and call in flow if not an external network